src/machine/board_waveshare-rp2040-zero.go

```diff
@@ -0,0 +1,129 @@
+//go:build waveshare_rp2040_zero
+// +build waveshare_rp2040_zero
+
+// This file contains the pin mappings for the Waveshare RP2040-Zero boards.
+//
+// Waveshare RP2040-Zero is a microcontroller using the Raspberry Pi RP2040 chip.
+//
+// - https://www.waveshare.com/wiki/RP2040-Zero
+package machine
+
+import (
+	"device/rp"
+	"runtime/interrupt"
+)
+
+// Digital Pins
+const (
+	D0  Pin = GPIO0
+	D1  Pin = GPIO1
+	D2  Pin = GPIO2
+	D3  Pin = GPIO3
+	D4  Pin = GPIO4
+	D5  Pin = GPIO5
+	D6  Pin = GPIO6
+	D7  Pin = GPIO7
+	D8  Pin = GPIO8
+	D9  Pin = GPIO9
+	D10 Pin = GPIO10
+	D11 Pin = GPIO11
+	D12 Pin = GPIO12
+	D13 Pin = GPIO13
+	D14 Pin = GPIO14
+	D15 Pin = GPIO15
+	D16 Pin = GPIO16
+	D17 Pin = GPIO17
+	D18 Pin = GPIO18
+	D19 Pin = GPIO19
+	D20 Pin = GPIO20
+	D21 Pin = GPIO21
+	D22 Pin = GPIO22
+	D23 Pin = GPIO23
+	D24 Pin = GPIO24
+	D25 Pin = GPIO25
+	D26 Pin = GPIO26
+	D27 Pin = GPIO27
+	D28 Pin = GPIO28
+	D29 Pin = GPIO29
+)
+
+// Analog pins
+const (
+	A0 Pin = D26
+	A1 Pin = D27
+	A2 Pin = D28
+	A3 Pin = D29
+)
+
+// Onboard LEDs
+const (
+	NEOPIXEL = GPIO16
+	LED      = NoPin
```

Comment:
If there is no LED, please leave the `LED` constant out. Having it there but defined as `NoPin` is only going to cause confusion.

Reply:
I'll do it tomorrow

Reply:
About testing it, i have it working with an USB device and connecting regular pins, but I haven't tested i2c, uart or spi yet. I have that pending

Reply:
@aykevl I removed the LED constant there. And I tested it with an I2C device that I have (an MP6050 accelerometer) and it looks like it's working fine. I don't have any UART or SPI based device at hand right now.